service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java

```diff
@@ -40,8 +56,49 @@ public UserDto findUserByIdService(Long id) {
         return userServiceMapper.convertUserToDto(user);
     }
 
+    @Transactional
     @Override
-    public Long count(){
+    public UserDto signup(UserDto userDto) {
+        User user = userServiceMapper.convertUserFromDto(userDto);
+
+        List<ValidationError> validationErrors = userValidator.validate(user);
```

Comment:
Is it advisable to do the conversion from DTO, and then validate? Why can't we immediately validate the object that we received from the request, and then, if necessary, validate it?

Reply:
fixed